lib/src/logger_extension.dart

```diff
@@ -11,13 +20,24 @@ extension LoggerX on Logger {
   /// Wrap the [text] to fit perfectly within the width of the terminal when
   /// [print]ed.
   ///
-  /// To overwrite the width you can use [length].
+  /// The text will wrap around the terminal width, and will not break words. If
+  /// the terminal width cannot be determined, the text will wrap around the
+  /// [fallbackStdoutTerminalColumns].
+  ///
+  /// To completely overwrite the width you can use [length].
   void wrap(
     String? text, {
     required void Function(String?) print,
     int? length,
   }) {
-    final maxLength = length ?? stdout.terminalColumns;
+    late final int maxLength;
+    try {
+      maxLength = length ?? stdout.terminalColumns;
+    } on StdoutException catch (_) {
+      // Not all terminals have a width, so we fallback to a reasonable value.
```

Comment:
It is only thrown if the terminal has no stdout, instead we should just check for [hasTerminal](https://api.dart.dev/stable/2.13.3/dart-io/Stdout/hasTerminal.html) and use that as the fallback.

Reply:
I amended the code. For other readers this information is documented within the `terminalColumns` method:

> If no terminal is attached to stdout, a [StdoutException] is thrown. See [hasTerminal] for more info.